Mathlib/LinearAlgebra/Contraction.lean

```diff
@@ -95,8 +95,9 @@ theorem map_dualTensorHom (f : Module.Dual R M) (p : P) (g : Module.Dual R N) (q
     TensorProduct.map (dualTensorHom R M P (f ⊗ₜ[R] p)) (dualTensorHom R N Q (g ⊗ₜ[R] q)) =
       dualTensorHom R (M ⊗[R] N) (P ⊗[R] Q) (dualDistrib R M N (f ⊗ₜ g) ⊗ₜ[R] (p ⊗ₜ[R] q)) := by
   ext m n
-  simp only [compr₂_apply, mk_apply, map_tmul, dualTensorHom_apply, dualDistrib_apply, ←
-    smul_tmul_smul]
+  simp only [AlgebraTensorModule.curry_apply, curry_apply, coe_restrictScalars, map_tmul,
+    dualTensorHom_apply, tmul_smul, dualDistrib_apply]
+  simp [← smul_tmul_smul]
```

Comment:
You just need `attribute [-ext] AlgebraTensorModule.curry_injective` in order to keep the original proofs (with `compr₂_apply` replaced by `compr₂ₛₗ_apply`).